deepset_cloud_sdk/workflows/async_client/files.py

```diff
@@ -54,6 +59,57 @@ async def list_files(
         pass
 
 
+async def list_upload_sessions(
+    api_key: Optional[str] = None,
+    api_url: Optional[str] = None,
+    workspace_name: str = DEFAULT_WORKSPACE_NAME,
+    is_expired: Optional[bool] = None,
+    batch_size: int = 100,
+    timeout_s: int = 300,
+) -> AsyncGenerator[List[UploadSessionDetail], None]:
+    """List all files in a workspace.
+
+    :param api_key: deepset Cloud API key to use for authentication.
+    :param api_url: API URL to use for authentication.
+    :param workspace_name: Name of the workspace to list the files from.
+    :param batch_size: Batch size for the listing.
+    :return: List of files.
+    """
+    try:
+        async with FilesService.factory(_get_config(api_key=api_key, api_url=api_url)) as file_service:
+            async for file_batch in file_service.list_upload_sessions(
+                workspace_name=workspace_name,
+                is_expired=is_expired,
+                batch_size=batch_size,
+                timeout_s=timeout_s,
+            ):
+                yield file_batch
+    except AsyncLibraryNotFoundError:
+        # since we are using asyncio.run() in the sync wrapper, we need to catch this error
```

Comment:
No. this is a hacky way to catch the async error if the user exits the context manager and the loop is not propperly closed :/ so this is not interesting for users, since they would get this also if everything is all right.